### ISSUE TITLE
the waveform disappear for longer files while zooming in

### ISSUE THREAD

**Issue by afaism** [katspaugh/wavesurfer.js#766]
Using http://codepen.io/katspaugh/pen/PZrBeE
I replaced the file name by http://ia902606.us.archive.org/35/items/shortpoetry_047_librivox/wantsofman_adams_jb_64kb.mp3 
When zooming in at some point the waveform disappears. 
Can you help?

Thanks!
Alex

**Comment by chrisparton1991** [comment]
Hi @afaism, this is a limitation with Canvas. On Chrome, a canvas cannot be larger than 32767 pixels. If it is (as is the case when you zoom in that file enough), the canvas will not render.

I implemented a `MultiCanvas` renderer a while back that overcomes this problem by stitching multiple canvases together horizontally. You as the user of the library don't have to worry about this beyond setting some wavesurfer parameters.

I have created a new fiddle that shows the changes: http://codepen.io/chrisparton1991/pen/XKgvgv

I upped the wavesurfer version to `1.1.1`, although you should use whatever the latest release is (I just know 1.1.1 has the MultiCanvas renderer). I also added a `renderer: 'MultiCanvas'` configuration parameter, and changed the URL to the one you specified.

Hopefully this should solve your issue :)

**Comment by afaisma** [comment]
Thanks as lot, chrisparton1991! 
This helped. However my next goal is to create a display of a 20+ min of media file with minimap/zoom and  hundreds of regions, 2 or more seconds each. The initial display of the waveform and the regions is fine, but when I trying to playback the media, the waveform disappears. 
( I posted this new question in this thread is because the disappearance of the waveform looks similar to the initial zoom problem). Following your suggestion I am using multicanvas . How expensive is the region object? If it is expensive, should i use a different approach?
Alex

**Comment by chrisparton1991** [comment]
No worries @afaisma, glad I could help :)

Unfortunately I'm just a guy who needed a multi-canvas renderer like you. I've implemented that and helped out with a couple of other things, but I've never used the regions plugin so I can't say what the best approach is there sorry!

@katspaugh, any recommendations?

**Comment by katspaugh** [comment]
Mhm, I'm not sure why the waveform would disappear after starting the playback.

Regions can be somewhat expensive if you say there are hundreds of them. Each creates a DOM element. I don't see how this is related to waveform disappearing though.

**Comment by chrisparton1991** [comment]
I've made some fixes to zooming recently that may (or may not) have an impact on this issue. I expect to raise another pull request over the next day or so with another fix. Once that is merged in, it would be good to get a CodePen example that reproduces this issue (if it still exists with my fixes).

I'll post on this issue again once my other fix is in.

**Comment by chrisparton1991** [comment]
The fixes have been merged in, just waiting on a new release to be made.

**Comment by katspaugh** [comment]
@chrisparton1991 thank you! Published version 1.1.11 on npm.

**Comment by chrisparton1991** [comment]
Awesome, thanks @katspaugh!

@afaism, could you please create a CodePen example with v1.1.11? As mentioned, this version has been released on npm. Cloudflare's CDN has only got release v1.1.10, but it should figure out that a new release has been made soon. The v1.1.11 file should be available from the following URL once the CDN has updated: https://cdnjs.cloudflare.com/ajax/libs/wavesurfer.js/1.1.11/wavesurfer.js.
